pkgs/applications/emulators/86box/default.nix

```diff
@@ -48,36 +49,42 @@ stdenv.mkDerivation rec {
     ++ lib.optional (!enableDynarec) "-DDYNAREC=OFF"
     ++ lib.optional (!unfreeEnableDiscord) "-DDISCORD=OFF";
 
-  postInstall = lib.optional stdenv.isLinux ''
+  postInstall = let
+    romsRepo = fetchFromGitHub {
+      owner = "86Box";
+      repo = "roms";
+      rev = "v${version}";
+      hash = "sha256-1HtoizO0QIGNjQTW0clzRp40h1ulw55+iTYz12UJSms=";
+    };
+  in (lib.optionalString stdenv.isLinux ''
```

Comment:
This is a bit better, but still a bit unreadable IMO, what I meant before is to use:

```nix
let
  romsRepo = ...
in stdenv.mkDerivation rec {
}
```

Another option, which is even better IMO, would be, to use:

```nix
stdenv.mkDerivation(finalAttrs: {
  ...
  passthru = {
    roms = fetchFromGitHub {
       ...
    };
  };

  postInstall = ''
  '' + lib.optionalString unfreeEnableRoms ''
    ln -s ${finalAttrs.passthru.roms} $out/...
  '';
```

This way, users who want to modify the roms derivation, could do it declarativly with:

```nix
my86box = _86box-with-roms.overrideAttrs:(finalAttrs: {
  passthru.roms = fetchFromGitHub {
    # Use my fork...
  };
};
```

I'm not totally sure about the above, so it is worth testing - you can insert that expression in `all-packages.nix` and try to build it and test it this way.

Reply:
nice

Reply:
I did the last one